src/tree.rs

```diff
@@ -376,6 +376,71 @@ impl<T> Default for Tree<T> {
     }
 }
 
+impl<T: std::fmt::Debug> Tree<T> {
+    /// Write formatted tree representation and nodes with debug formatting.
+    ///
+    /// Example:
+    ///
+    /// ```
+    /// use slab_tree::tree::TreeBuilder;
+    ///
+    /// let mut tree = TreeBuilder::new().with_root(0).build();
+    /// let mut root = tree.root_mut().unwrap();
+    /// root.append(1)
+    ///     .append(2);
+    /// root.append(3);
+    /// let mut s = String::new();
+    /// tree.write_formatted(&mut s).unwrap();
+    /// assert_eq!(&s, "\
+    /// 0
+    /// ├── 1
+    /// │   └── 2
+    /// └── 3
+    /// ");
+    /// ```
+    pub fn write_formatted<W: std::fmt::Write>(&self, w: &mut W) -> std::fmt::Result {
+        let root = self.root().unwrap();
```

Comment:
I believe this line will panic if there's no root node in the tree.  Do you mind dealing with that condition here?
I was thinking something along the lines of:
```
if let Some(root) = self.root() {
    return write_node(...);
}
Ok(())
```
But there may be other better ways to handle that.